app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntryUpdate.kt

```diff
@@ -63,6 +66,15 @@ class BloodPressureEntryUpdate : Update<BloodPressureEntryModel, BloodPressureEn
 
       is YearChanged -> dispatch(HideDateErrorMessage)
 
+      is BloodPressureDateClicked -> {
+        val result = bpValidator.validate(model.systolic, model.diastolic)
```

Comment:
Suggestion: Convert the if statement into a `when` block and inline the `result` into the `when` subject.

```kotlin
is BloodPressureDateClicked -> when (val result = bpValidator.validate(model.systolic, model.diastolic)) {
  is Success -> dispatch<BloodPressureEntryModel, BloodPressureEntryEffect>(ShowDateEntryScreen)
  else -> dispatch<BloodPressureEntryModel, BloodPressureEntryEffect>(ShowBpValidationError(result))
}

```

Reply:
Same as above, when it comes to Mobius classes.